src/hammer-vlsi/hammer_vlsi/cli_driver.py

```diff
@@ -57,16 +57,45 @@ def dump_config_to_json_file(output_path: str, config: dict) -> None:
         f.write(json.dumps(config, indent=4))
 
 
-# Type signature of a CLIDriver action.
-CLIActionType = Callable[[HammerDriver, Callable[[str], None]], Optional[dict]]
+# Type signature of a CLIDriver action that returns a config dictionary.
+CLIActionConfigType = Callable[[HammerDriver, Callable[[str], None]], Optional[dict]]
+
+# Type signature of a CLIDriver action that returns a raw string.
+CLIActionStringType = Callable[[HammerDriver, Callable[[str], None]], Optional[str]]
+
+# CLIDriver action types.
+CLIActionType = Union[CLIActionConfigType, CLIActionStringType]
+
+
+# We cannot use isinstance() with CLIActionConfigType directly:
+# "Parameterized generics cannot be used with class or instance checks"
+
+def is_config_action(func: CLIActionType) -> bool:
+    """Return True if the given function is a CLIActionConfigType."""
+    return check_function_type(func, [HammerDriver, Callable[[str], None]], Optional[dict]) is None
+
+
+def is_string_action(func: CLIActionType) -> bool:
+    """Return True if the given function is a CLIActionConfigType."""
+    return check_function_type(func, [HammerDriver, Callable[[str], None]], Optional[str]) is None
 
 
 def check_CLIActionType_type(func: CLIActionType) -> None:
     """
-    Check that the given CLIActionType obeys its function type signature.
+    Check that the given CLIActionConfigType obeys its function type signature.
```

Comment:
This change and the name of the function seem unaligned. It looks like its checking any CLIActionTypes